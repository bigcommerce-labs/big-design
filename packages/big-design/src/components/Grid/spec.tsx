import 'jest-styled-components';
import React from 'react';
import { render } from 'react-testing-library';

import { Grid } from './index';

test('render Grid', () => {
  const template = `
  "head head" 80px
  "nav  main" 1fr
  "nav  foot" 10%
  / 120px 1fr;
`;

  const { container } = render(
    <Grid template={template}>
      <Grid.Item area="head">Header</Grid.Item>
      <Grid.Item area="nav">Sidebar</Grid.Item>
      <Grid.Item area="main">Content</Grid.Item>
      <Grid.Item area="foot">Footer</Grid.Item>
    </Grid>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('has display grid', () => {
  const { container } = render(<Grid />);

  expect(container.firstChild).toHaveStyle('display: grid');
});

test('Grid forwards styles', () => {
  const { container } = render(<Grid className="test" style={{ background: 'red' }} />);

  expect(container.getElementsByClassName('test').length).toBe(1);
  expect(container.firstChild).toHaveStyle('background: red');
});

test('Grid item forwards styles', () => {
  const { container } = render(<Grid.Item className="test" style={{ background: 'red' }} />);

  expect(container.getElementsByClassName('test').length).toBe(1);
  expect(container.firstChild).toHaveStyle('background: red');
});
