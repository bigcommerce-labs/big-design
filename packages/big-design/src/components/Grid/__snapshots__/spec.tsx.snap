// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Grid 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  grid-gap: 1rem;
  grid-template: "head head" 80px "nav  main" 1fr "nav  foot" 10% / 120px 1fr;
  display: grid;
}

.c2 {
  grid-area: head;
}

.c3 {
  grid-area: nav;
}

.c4 {
  grid-area: main;
}

.c5 {
  grid-area: foot;
}

<div
  class="c0 c1"
>
  <div
    class="c0 c2"
  >
    Header
  </div>
  <div
    class="c0 c3"
  >
    Sidebar
  </div>
  <div
    class="c0 c4"
  >
    Content
  </div>
  <div
    class="c0 c5"
  >
    Footer
  </div>
</div>
`;
